File: app/src/main/java/com/mayan/numbers/decimal/DrawNumbersView.java
package com.mayan.numbers.decimal;

import java.util.Hashtable;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.os.Bundle;
import android.view.View;

public class DrawNumbersView extends View {

	int[] numbers;
	private Hashtable<String, Bitmap> numberHash = new Hashtable<String,Bitmap>();
	
	public DrawNumbersView(Context context) {
		super(context);
		
		numberHash.put( String.valueOf(0) , BitmapFactory.decodeResource(getResources(), R.drawable.n0) );
		numberHash.put( String.valueOf(1) , BitmapFactory.decodeResource(getResources(), R.drawable.n1) );
		numberHash.put( String.valueOf(2) , BitmapFactory.decodeResource(getResources(), R.drawable.n2) );
		numberHash.put( String.valueOf(3) , BitmapFactory.decodeResource(getResources(), R.drawable.n3) );
		numberHash.put( String.valueOf(4) , BitmapFactory.decodeResource(getResources(), R.drawable.n4) );
		numberHash.put( String.valueOf(5) , BitmapFactory.decodeResource(getResources(), R.drawable.n5) );
		numberHash.put( String.valueOf(6) , BitmapFactory.decodeResource(getResources(), R.drawable.n6) );
		numberHash.put( String.valueOf(7) , BitmapFactory.decodeResource(getResources(), R.drawable.n7) );
		numberHash.put( String.valueOf(8) , BitmapFactory.decodeResource(getResources(), R.drawable.n8) );
		numberHash.put( String.valueOf(9) , BitmapFactory.decodeResource(getResources(), R.drawable.n9) );
		numberHash.put( String.valueOf(10) , BitmapFactory.decodeResource(getResources(), R.drawable.n10) );
		numberHash.put( String.valueOf(11) , BitmapFactory.decodeResource(getResources(), R.drawable.n11) );
		numberHash.put( String.valueOf(12) , BitmapFactory.decodeResource(getResources(), R.drawable.n12) );
		numberHash.put( String.valueOf(13) , BitmapFactory.decodeResource(getResources(), R.drawable.n13) );
		numberHash.put( String.valueOf(14) , BitmapFactory.decodeResource(getResources(), R.drawable.n14) );
		numberHash.put( String.valueOf(15) , BitmapFactory.decodeResource(getResources(), R.drawable.n15) );
		numberHash.put( String.valueOf(16) , BitmapFactory.decodeResource(getResources(), R.drawable.n16) );
		numberHash.put( String.valueOf(17) , BitmapFactory.decodeResource(getResources(), R.drawable.n17) );
		numberHash.put( String.valueOf(18) , BitmapFactory.decodeResource(getResources(), R.drawable.n18) );
		numberHash.put( String.valueOf(19) , BitmapFactory.decodeResource(getResources(), R.drawable.n19) );
		
	}
	
	public void draw(Canvas canvas) {
		super.draw(canvas);
		try {
            Bitmap backgroundImage = BitmapFactory.decodeResource( getResources() , R.drawable.fondo_chichen_2  );
            canvas.drawBitmap(backgroundImage, 0 , 0 , null );
            int count = 0;
            for (int i = 0; i < numbers.length; i++) {
                Bitmap image = numberHash.get( String.valueOf(numbers[i]) );
                canvas.drawBitmap( image , canvas.getWidth()/2 - image.getWidth()/2 , (count * image.getHeight() )+5 , null);
                count++;
            }
        }catch(Exception ex) {
            ex.printStackTrace();
            showErrorMessageInMain();
        }
	}

	public void setNumbersToDraw(int[] numbers) {
		this.numbers = numbers;
	}

    public void showErrorMessageInMain(){
        Bundle extras = new Bundle();
        extras.putInt( "errMsg" , R.string.error_when_converting_big_number );
        Intent intent = new Intent( (Activity)getContext() , MainActivity.class );
        intent.putExtras(extras);
        getContext().startActivity(intent);
    }



}

File: app/src/main/java/com/mayan/numbers/decimal/DrawNumbersView.kt
package com.mayan.numbers.decimal

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.Bundle
import android.content.Intent
import android.app.Activity
import android.content.Context
import android.graphics.Canvas
import android.view.View
import java.lang.Exception
import java.util.Hashtable


class DrawNumbersView(context: Context?) : View(context) {
    var numbers: IntArray? = null
    private val numberHash = Hashtable<String, Bitmap>()
    override fun draw(canvas: Canvas) {
        super.draw(canvas)
        try {
            val backgroundImage =
                BitmapFactory.decodeResource(resources, R.drawable.fondo_chichen_2)
            canvas.drawBitmap(backgroundImage, 0f, 0f, null)
            for ((count, number) in numbers!!.withIndex()) {
                val image = numberHash[number.toString()]
                canvas.drawBitmap(
                    image!!,
                    (width / 2 - image.width / 2).toFloat(),
                    (count * image.height + 5).toFloat(),
                    null
                )
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
            showErrorMessageInMain()
        }
    }

    fun setNumbersToDraw(numbers: IntArray?) {
        this.numbers = numbers
    }

    private fun showErrorMessageInMain() {
        val extras = Bundle()
        extras.putInt("errMsg", R.string.error_when_converting_big_number)
        val intent = Intent(context as Activity, MainActivity::class.java)
        intent.putExtras(extras)
        context.startActivity(intent)
    }

    init {
        numberHash[0.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n0)
        numberHash[1.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n1)
        numberHash[2.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n2)
        numberHash[3.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n3)
        numberHash[4.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n4)
        numberHash[5.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n5)
        numberHash[6.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n6)
        numberHash[7.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n7)
        numberHash[8.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n8)
        numberHash[9.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n9)
        numberHash[10.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n10)
        numberHash[11.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n11)
        numberHash[12.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n12)
        numberHash[13.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n13)
        numberHash[14.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n14)
        numberHash[15.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n15)
        numberHash[16.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n16)
        numberHash[17.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n17)
        numberHash[18.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n18)
        numberHash[19.toString()] = BitmapFactory.decodeResource(resources, R.drawable.n19)
    }
}